ultralytics/utils/checks.py

```diff
@@ -163,31 +164,41 @@ def check_version(current: str = '0.0.0',
 
         # check if current version is between 20.04 (inclusive) and 22.04 (exclusive)
         check_version(current='21.10', required='>20.04,<22.04')
+        ```
     """
     if not current:  # if current is '' or None
         LOGGER.warning(f'WARNING ⚠️ invalid check_version({current}, {required}) requested, please check values.')
         return True
+    elif not current[0].isdigit():  # current is package name rather than version string, i.e. current='ultralytics'
+        try:
+            name = current  # assigned package name to 'name' arg
+            current = version(current)  # get version string from package name
+        except PackageNotFoundError:
+            if hard:
+                raise ModuleNotFoundError(emojis(f'WARNING ⚠️ {current} package is required but not installed'))
+            else:
+                return False
 
     if not required:  # if required is '' or None
         return True
 
-    current = parse_version(current)  # '1.2.3' -> (1, 2, 3)
+    c = parse_version(current)  # '1.2.3' -> (1, 2, 3)
     constraints = re.findall(r'([<>!=]{1,2}\s*\d+\.\d+)', required) or [f'>={required}']
     result = True
     for constraint in constraints:
-        op, v = re.match(r'([<>!=]{1,2})\s*(\d+\.\d+)', constraint).groups()
+        op, v = re.match(r'([^0-9]*)([\d.]+)', constraint).groups()
```

Comment:
hmm, why changing it this way? should it still include the `op` part - `[<>!=]` at least? 

it seems it should be the same as in the ` constraints = re.findall` above.

if anything, I would probably try to even make a constant so that both are in sync

Reply:
Yeah this is a good point. Let me see what i can do here.

Reply:
@shcheklein ah ok, these are a bit different, as the first is just supposed to parse constraints into a list, and the second to parse the inequalities from the version number:

constraints = re.findall(r'([<>!=]{1,2}\s*\d+\.\d+)', '>=1.9.0,<=1.8.0') or [f'>={required}']
constraints
Out[11]: ['>=1.9', '<=1.8']

Then second match further splits them, so the patterns can be slightly different.

Reply:
@shcheklein eliminated the first re.findall call completely now 
 in https://github.com/ultralytics/ultralytics/pull/5068/commits/34c057247fa10f776abb238b90a3ee870942edef
 
 This is a bit simpler, and only 1 pattern now. If this looks good I'll go ahead and merge, and should be part of 8.0.186 tomorrow.

Reply:
looks good to me. I've tested the branch manually and it works good. Thanks @glenn-jocher for the quick fix!